### PR TITLE
fix database query invoker to be async

### PR DESCRIPTION
**Pull request by gadomsky**

### CODE REVIEW

File: components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvoker.scala [hunk 1]
```diff
@@ -30,16 +30,15 @@ class DatabaseEnricherInvoker(query: String,
...
                             (implicit ec: ExecutionContext, collector: ServiceInvocationCollector, contextId: ContextId, componentUseCase: ComponentUseCase): Future[queryExecutor.QueryResult] =
     getTimeMeasurement().measuring {
-      Future.successful {
-        queryDatabase(queryArgumentsExtractor(argsCount, params))
-      }
+      queryDatabase(queryArgumentsExtractor(argsCount, params))
     }
 
-  protected def queryDatabase(queryArguments: QueryArguments): queryExecutor.QueryResult =
+  protected def queryDatabase(queryArguments: QueryArguments)(implicit ec: ExecutionContext): Future[queryExecutor.QueryResult] = Future {
```
Comment (arkadius) [inline components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvoker.scala:36]: Can you move this `Future {` to separate line? I don't know if only I have that, but for me, this Future vanish on the end of line.

File: components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvokerWithCache.scala [hunk 2]
```diff
@@ -33,19 +33,20 @@ class DatabaseEnricherInvokerWithCache(query: String,
   // TODO: cache size
   private val cache = Caffeine.newBuilder()
     .expireAfterWrite(cacheTTL)
-    .build[CacheKey, CacheEntry[queryExecutor.QueryResult]]
+    .buildAsync[CacheKey, CacheEntry[queryExecutor.QueryResult]]
+
+  import scala.jdk.FutureConverters._
```
Comment (arkadius) [inline components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvokerWithCache.scala:38]: You need to add some compat lib for that - it won't work on 2.12

File: components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvokerWithCache.scala [hunk 3]
```diff
@@ -33,19 +33,20 @@ class DatabaseEnricherInvokerWithCache(query: String,
   // TODO: cache size
   private val cache = Caffeine.newBuilder()
     .expireAfterWrite(cacheTTL)
-    .build[CacheKey, CacheEntry[queryExecutor.QueryResult]]
+    .buildAsync[CacheKey, CacheEntry[queryExecutor.QueryResult]]
```
Comment (dswiecki) [inline components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvokerWithCache.scala:36]: maybe lets extend some Tests just to be sure that caching works when is should

Comment (gadomsky) [reply components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvokerWithCache.scala:36]: it's actually testing a library.. but I've added test proving that it works

Comment (gadomsky) [reply components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvokerWithCache.scala:36]: + it was already tested though - we do update on DB, and checking that enricher returns old value

Comment (gadomsky) [reply components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseEnricherInvokerWithCache.scala:36]: or maybe you mean testing other behaviour? if so, which one?

**Review by arkadius — APPROVED** [review]
LGTM

**Pull request merged**